borg/archiver.py

```diff
@@ -761,6 +761,19 @@ def build_parser(self, args=None, prog=None):
         See the output of the "borg help patterns" command for more help on exclude patterns.
         """)
 
+        class ToggleAction(argparse.Action):
+            """argparse action to handle "toggle" flags easily
+
+            toggle flags are in the form of ``--foo``, ``--no-foo``.
+
+            the ``--no-foo`` argument still needs to be passed to the
+            ``add_argument()`` call, but it simplifies the ``--no``
+            detection.
+            """
+            def __call__(self, parser, ns, values, option):
+                """set the given flag to true unless ``--no`` is passed"""
+                setattr(ns, self.dest, option.startswith('--no-'))
+
```

Comment:
i tried moving it up before build_args(), but then it can't be found in the scope and looks weird. i could move it up the base Archiver class, but then that would be the only other class here.

it would be great to decouple that module, it's getting more and more ugly.